cmd/infra/aws/create.go

```diff
@@ -19,6 +19,8 @@ type CreateInfraOptions struct {
 	Region             string
 	InfraID            string
 	AWSCredentialsFile string
+	AWSKey             string
+	AWSSecretKey       string
```

Comment:
should there be flags for these? Also if there are, AWSCreds should not be required (https://github.com/openshift/hypershift/pull/860/files#diff-718f3a535d5069f65c655a6143748ec6b3ac901d75f86e8ce59fe79fad4078ceR77). You may also want to add some validation that you can specify either/or but not both.

Reply:
That's the question I had. I don't know if we want to add two flags per provider for credentials.  All three main cloud providers have the local file.  The reason this was introducing this was if you want to call infra.aws.create or infra.aws.create_iam from external go code, in a controller, there is no way to have the creds file, but now you can dynamically inject the static cred's in this usecase.

If we feel we want to add flags for these though, I can add them in.

@csrwng does t

Reply:
I think I'd rather avoid to enable users to pass text plain creds through the cli, specially since we don't have a need for it atm.

Reply:
@enxebre agreed. I didn't initially understand that this code is being vendored and called directly. That being the case, I see no need to expose externally.